src/libs/ajax/User.ts

```diff
@@ -0,0 +1,398 @@
+        );
+      },
+
+      setPreferences: async (body: {}): Promise<void> => {
+        return fetchOrchestration(
+          'api/profile/preferences',
+          _.mergeAll([authOpts(), jsonBody(body), { signal, method: 'POST' }])
+        );
+      },
+
+      preferLegacyFirecloud: async (): Promise<OrchestrationUserPreferLegacyFireCloudResponse> => {
+        return fetchOrchestration('api/profile/terra', _.mergeAll([authOpts(), { signal, method: 'DELETE' }]));
+      },
+    },
+
+    // Returns the proxy group email of the user with the given email
+    getProxyGroup: async (email: string): Promise<string> => {
+      const res = await fetchOrchestration(
+        `api/proxyGroup/${encodeURIComponent(email)}`,
+        _.merge(authOpts(), { signal })
+      );
+      return res.json();
+    },
+
+    getTos: async (): Promise<string> => {
+      const response = await fetchSam('tos/text', _.merge(authOpts(), { signal }));
+      return response.text();
+    },
+
+    acceptTos: async (): Promise<SamUserTosStatusResponse | undefined> => {
+      try {
+        const response = await fetchSam(
+          'register/user/v1/termsofservice',
+          _.mergeAll([authOpts(), { signal, method: 'POST' }, jsonBody('app.terra.bio/#terms-of-service')])
+        );
+        return response.json();
+      } catch (error: any) {
+        if (error.status !== 404) {
+          throw error;
+        }
+      }
+    },
+
+    rejectTos: async (): Promise<SamUserTosStatusResponse | undefined> => {
+      try {
+        const response = await fetchSam(
+          'register/user/v1/termsofservice',
+          _.mergeAll([authOpts(), { signal, method: 'DELETE' }])
+        );
+        return response.json();
+      } catch (error: any) {
+        if (error.status !== 404) {
+          throw error;
+        }
+      }
+    },
+
+    getTermsOfServiceComplianceStatus: async (): Promise<SamUserTosComplianceStatusResponse | null> => {
+      try {
+        const res = await fetchSam(
+          'register/user/v2/self/termsOfServiceComplianceStatus',
+          _.merge(authOpts(), { signal })
+        );
+        return res.json();
+      } catch (error: any) {
+        if (error.status === 404 || error.status === 403) {
+          return null;
+        }
+        throw error;
+      }
+    },
+
+    getPrivacyPolicy: async (): Promise<string> => {
+      const response = await fetchSam('privacy/text', _.merge(authOpts(), { signal }));
+      return response.text();
+    },
+
+    // If you are making changes to the Support Request Modal, make sure you test the following:
+    // 1. Submit a ticket via Terra while signed in and signed out
+    // 2. Check the tickets are generated on Zendesk
+    // 3. Reply internally (as a Light Agent) and make sure an email is not sent
+    // 4. Reply externally (ask one of the Comms team with Full Agent access) and make sure you receive an email
+    createSupportRequest: ({
```

Comment:
Were types for `createSupportRequest` and `uploadAttachment` purposefully left out?

Reply:
Not exactly, I was not sure where to find their documentation. Will ask around.

Reply:
Following up that these are Zendesk API endpoints https://developer.zendesk.com/api-reference/ticketing/tickets/ticket-requests/#create-request and https://developer.zendesk.com/documentation/ticketing/using-the-zendesk-api/adding-ticket-attachments-with-the-api/ ty @nawatts